determined-ai/determined
# feat: add auto namespace and resource quota inputs to webui

## Ticket

DET-10284


## Description




## Test Plan

Non Kubernetes RM:
1. Verify when you go to Create a Workspace or Edit a workspace from the UI, there is no place to create workspace namespace Bindings.

EE Kubernetes RM:
1. Create a workspace first without the auto-create toggle on and nothing in the namespace input.
2. Try to edit the same workspace, it should pre-populate the same input with "default" or the default namespace configured in the master config. 
3. Change the namespace to a valid namespace from that cluster. Save and it should save without errors and verify it shows the updated namespace in that input box when you try to edit it again.
4. Edit the workspace again, and turn the au to-create namespace toggle on.
5. Verify that the namespace input gets greyed out, and an input for resource quota pops up. Add an integer input for resource quota and save.
6. Verify that it displays a new autocreated namespace name in the namespace input field and the resource quota populates correctly.
7. Change the resource quota and verify it updates correctly after saving.
8. Turn the auto create toggle off and verify it goes back to the default namespace after saving.
9. Verify that the resource quota field is not editable when auto-create namespace toggle is off.
10. Create a namespace in kubernetes and add a resource quota of 5 GPUS to it using kubectl.
11. bind the namespace to the workspace using the webui. Verify that the resource quota of 5 populates correctly after saving.
12. Now add another resource quota of 3 to the same namespace. verify that now the resource quota of 3 shows up.
13. Add another resource quota of 7 to the same namespace and verify it still shows as 3 in the webui

OSS Kubernetes RM:
- Verify that the auto create toggle does not show up but a field for inputting namespace shows up in the create workspace modal.

## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.




[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
